core/src/main/java/org/apache/calcite/rel/rules/SortJoinTransposeRule.java

```diff
@@ -181,4 +190,17 @@ default Config withOperandFor(Class<? extends Sort> sortClass,
           .as(Config.class);
     }
   }
+
+  @Nullable private RexNode newLimit(RexBuilder rexBuilder,
+      @Nullable RexNode offset, @Nullable RexNode limit) {
+    if (offset == null || limit == null) {
+      return limit;
+    }
+    final BigDecimal offsetValue = ((RexLiteral) offset).getValueAs(BigDecimal.class);
+    final BigDecimal limitValue = ((RexLiteral) limit).getValueAs(BigDecimal.class);
```

Comment:
is it safe to cast to `RexLiteral` or should I add a `instanceof` check? Are there cases where `offset`/`fetch` may not be a `RexLiteral`?